src/Illuminate/Database/Migrations/Migrator.php

```diff
@@ -557,7 +568,9 @@ public function getMigrationFiles($paths)
     public function requireFiles(array $files)
     {
         foreach ($files as $file) {
-            $this->files->requireOnce($file);
+            if (! isset(static::$requiredMigrations[$file])) {
+                static::$requiredMigrations[$file] = $this->files->requireOnce($file);
```

Comment:
If I remember correctly, require_once only returns `true` if the file was successfully required and does not return the return value of the file. That might be unintended at this stage